nginx/parts/plugins/x-autotools.py

```diff
@@ -0,0 +1,62 @@
+"""This is a modified version of snapcraft's autotools plugin.
+
+It calls nginx' ./auto/configure
+
+    - configflags:
+      (list of strings)
+      configure flags to pass to the build such as those shown by running
+      './auto/configure --help'
+"""
+
+import snapcraft
+
+
+class AutotoolsPlugin(snapcraft.BasePlugin):
+
+    @classmethod
+    def schema(cls):
+        schema = super().schema()
+        schema['properties']['configflags'] = {
+            'type': 'array',
+            'minitems': 1,
+            'uniqueItems': True,
+            'items': {
+                'type': 'string',
+            },
+            'default': [],
+        }
+
+        # Inform Snapcraft of the properties associated with building. If these
+        # change in the YAML Snapcraft will consider the build step dirty.
```

Comment:
ugh, we need to change that in all the plugins :(
maybe a bitesize bug to teach somebody new how to use git and start contributing.